spec/views/images/crop.html.erb_spec.rb

```diff
@@ -0,0 +1,38 @@
+RSpec.describe "images/crop.html.erb" do
+  describe "image crop" do
+    it "shows a 'Save' button by with JS" do
+      assign(:image_revision, create(:image_revision))
+      assign(:edition, build(:edition))
+      render
+      expect(rendered).to have_button("Save")
+    end
+
+    it "shows a 'Continue' button without JS" do
+      assign(:image_revision, create(:image_revision))
+      assign(:edition, build(:edition))
+      render
+      expect(rendered).to have_selector('.app-no-js button[text()="Continue"]')
```

Comment:
I think this would be cleaner as:

```ruby
      expect(rendered).to have_selector(".app-no-js button", "Continue")
```

I don't think I've come across the `text()` syntax before 🤔 is that an RSpec thing, a Capybara thing or a CSS3 thing?

Reply:
Turns out it's an xpath thing - I'm used to that kind of syntax [from govuk_publishing_components](https://github.com/alphagov/govuk_publishing_components/blob/master/spec/components/file_upload_spec.rb#L53), although your suggestion is a bit nicer. Updated.